Structure_Engine/Compute/PanelToFEMesh.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using BH.oM.Structure.Elements;
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.Engine.Analytical;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Converting panel with 3 or 4 control points to FEmesh ")]
+        [Input("panel", "Panel to be converted to an FEMesh.")]
+        [Output("FEMesh", "FEMesh converted from a Panel.")]
+
+        public static FEMesh PanelToFEMesh(Panel panel)
+        {
+            List<Point> points = new List<Point>();
+            List<Edge> edges = panel.ExternalEdges;
+            List<Face> faces = new List<Face>();
+            if (panel.Openings.Count()>0)
+            {
+                Engine.Reflection.Compute.RecordWarning("PanelToFEMesh does not support panel with opening");
+                return null;
+            }
+            if (panel.ExternalEdges.Count>4)
+            {
+                Engine.Reflection.Compute.RecordWarning("Panel contains more than 4 control points");
+                return null;
+            }
```

Comment:
You need to add some additional checks here, because you could have an `Edge` that is defined by a single `Polyline` with multiple control points. Whereas your error message refers to control points.